hpat/tests/test_series.py

```diff
@@ -417,76 +417,106 @@ def test_impl(A):
             hpat_func(S), test_impl(S).reset_index(drop=True))
 
     def test_series_op1(self):
-        def test_impl(A, i):
-            return A + A
-        hpat_func = hpat.jit(test_impl)
+        arithmetic_binops = ('+', '-', '*', '/', '//', '%', '**')
 
-        n = 11
-        df = pd.DataFrame({'A': np.arange(n)})
-        pd.testing.assert_series_equal(hpat_func(df.A, 0),
-                                       test_impl(df.A, 0), check_names=False)
+        for operator in arithmetic_binops:
+            func_text = "def test_impl(S):\n"
+            func_text += "   return S {} S\n".format(operator)
+            loc_vars = {}
+            exec(func_text, {}, loc_vars)
+            test_impl = loc_vars['test_impl']
+            hpat_func = hpat.jit(test_impl)
+        
+            n = 11
+            df = pd.DataFrame({'A': np.arange(1, n)})
+            pd.testing.assert_series_equal(hpat_func(df.A), test_impl(df.A), check_names=False)
 
-    @unittest.skip('AssertionError - fix needed\n'
-                   'Attribute "dtype" are different\n'
-                   '[left]:  int64\n'
-                   '[right]: int32\n')
     def test_series_op2(self):
-        def test_impl(A, i):
-            return A+i
-        hpat_func = hpat.jit(test_impl)
+        arithmetic_binops = ('+', '-', '*', '/', '//', '%', '**')
 
-        n = 11
-        df = pd.DataFrame({'A': np.arange(n)})
-        pd.testing.assert_series_equal(hpat_func(df.A, 1),
-                                       test_impl(df.A, 1), check_names=False)
+        for operator in arithmetic_binops:
+            func_text = "def test_impl(S, i):\n"
+            func_text += "   return S {} i\n".format(operator)
```

Comment:
Is it possible to avoid autogeneration of the code and `exec()` method?

Reply:
Not sure if decorator-solution works here, as we use operators in the function body.
Will check, but I don't really think it's a problem, because it's not in functional code, but in tests and they seem to be simpler with this form.